vernac/comAssumption.ml

```diff
@@ -173,6 +173,7 @@ let do_assumptions ~pstate ~program_mode kind nl l =
       uvars, (coe,t,imps))
       Univ.LSet.empty l
   in
+  (* XXX: Should adapt to use DeclareDef.prepare_parameter *)
```

Comment:
Doesn't really seem possible considering we have to produce multiple parameters.

Reply:
I was thinking of adapting `prepare_parameter` to handle this case, I ignore how feasible it is.

Reply:
I think we're better off keeping it simple and doing a separate implementation when we want something more complex.

Reply:
For this case you are likely right; but in general, duplication in the proof handling path is a serious problem in the codebase and we _must_ abstract out common functionality. Pity I cannot submit a few PRs to illustrate why I mean :(

Reply:
Ideally we would even use typing to enforce clients of the API to pass thru the correct library functions, a part of this is done already in my proof-tree.

Reply:
>For this case you are likely right

Please remove the comment then.

Reply:
I still think this comment should go.

Reply:
My plan was to improve prepare parameter so it could handle this case. Would it be OK for you if I improve the comment to reflect that?

[I'd really like to de-duplicate here]

Reply:
>Would it be OK for you if I improve the comment to reflect that?

I can't answer that without seeing the improved comment. (I won't fight if you insist on having the comment though)

Reply:
I added a new version; I don't mind either removing the commend if you don't like it; but such comments tend to be useful for me tho.